### ISSUE TITLE
Pager/paging is not working when focus is in "Unstaged Changes" or "Staged Changes" window

### ISSUE THREAD

**Issue by c02y** [jesseduffield/lazygit#1939]
**Describe the bug**

**To Reproduce**
1. open a git repo using lazygit
2. switch focus to a modified file in `Files` side panel
3. check the `Unstaged Changed` and `Staged Changes`, the pager is working fine
4. `Enter` on the file, the focus will be in the `Staged Changes`, the pager is not working now

**Expected behavior**
pager will work when focus is in the  "Unstaged Changes" or "Staged Changes" window

**Screenshots**
![image](https://user-images.githubusercontent.com/2135996/167940701-15f254b2-7856-437b-8eb4-1ede47b76310.png)

![image](https://user-images.githubusercontent.com/2135996/167940759-d42684c4-b97e-4769-8630-f0dc837c3496.png)

**Desktop (please complete the following information):**
 - OS: [e.g. Windows] `Arch Linux 5.17.5-zen1-1-zen`
 - Lazygit Version [e.g. v0.1.45] `v0.34` installed from Arch Linux official repo
 - The last commit id if you built project from sources (run : ```git rev-parse HEAD```)

**Additional context**

My related config:
```
git:
  paging:
    colorArg: always
    useConfig: false
    pager: delta --paging=never --line-numbers
```

**Comment by mark2185** [comment]
It seems there are issues with the `delta` pager. 

If this is still a problem, please consider switching until it gets fixed.

**Comment by c02y** [comment]
Not really, I tried `diff-so-fancy` (use config from https://github.com/jesseduffield/lazygit/blob/master/docs/Custom_Pagers.md） , it doesn't show colors of diffs as well when I switch focus into `Unstaged/Staged Changes` windows, and more importantly, lazygit crashed like 100% when I exit from this focus using `Escape`

```
$ lazygit --version
commit=v0.35, build date=2022-07-20T10:20:42Z, build source=binaryRelease, version=0.35, os=linux, arch=amd64

$ lazygit
2022/08/05 15:59:23 An error occurred! Please create an issue at: https://github.com/jesseduffield/lazygit/issues

syscall.Errno bad file descriptor
github.com/jesseduffield/lazygit/pkg/app/app.go:52 (0x5620b5fce736)
github.com/jesseduffield/lazygit/main.go:154 (0x5620b5fdc190)
runtime/proc.go:250 (0x5620b59c97d3)
runtime/asm_amd64.s:1571 (0x5620b59f7261)
```

I tried `ydiff`  (use config from https://github.com/jesseduffield/lazygit/blob/master/docs/Custom_Pagers.md), it doesn't display colors like the demo screenshot at all.
![image](https://user-images.githubusercontent.com/2135996/183032195-32b56c14-f242-4f7d-be59-48a30701c9fa.png)


So I think there are a lot of issues for custom_pager code in lazygit.
